src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -137,6 +137,13 @@ protected HandleReferencesBase(Index index, DocumentsStorage documentsStorage, I
                 if (lastIndexedEtag == 0) // we haven't indexed yet, so we are skipping references for now
                     continue;
 
+                // When opening the read transaction for a single map covering all of its referenced collections,
+                // we can efficiently skip indexing documents that have already been indexed.
+                var readTransaction = queryContext.OpenReadTransaction();
+                var indexed = new HashSet<Slice>(SliceComparer.Instance);
```

Comment:
Previously we had `OrdinalIgnoreCase` while the `SliceComparer` doesn't have the notion of ignoring the casing. Do we care about that?

Reply:
It doesn't matter. Those are the parent item IDs which are the same across different referenced collections.